### PR TITLE
Enhance PII masking regex tests and fix failing workflows

### PR DESCRIPTION
**Pull request by IsuruGunarathne**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

## Summary by CodeRabbit

* **New Features**
  * Added request capture and a new endpoint to retrieve the last captured request.

* **Tests**
  * Expanded PII-masking tests with comprehensive assertions across emails, phones, SSNs, credit cards, nested JSON paths, multi-PII scenarios, and edge cases; added capture/response validation.

* **Documentation**
  * Updated README with examples showing how to capture and retrieve requests.

* **Chores**
  * Bumped CI Go minor patch version across workflows.

### CODE REVIEW

**Review by renuka-fernando — DISMISSED** [review]

**Review by Tharsanan1 — APPROVED** [review]

File: .github/workflows/trivy-scan.yaml [hunk 1]
```diff
@@ -70,7 +70,7 @@ jobs:
 
       - name: Build & push router
         run: |
-          make -C gateway/router build-and-push-multiarch \
+          make -C gateway/gateway-runtime/router build-and-push-multiarch \
           IMAGE_NAME=localhost:5000/router \
           VERSION=trivy
```
Comment (renuka-fernando) [inline .github/workflows/trivy-scan.yaml]: We don't need this step, because now there is no router docker image

File: .github/workflows/trivy-scan.yaml
Comment (renuka-fernando) [inline .github/workflows/trivy-scan.yaml]: Should be Gateway Runtime
```suggestion
      - name: Build & push gateway-runtime image
```

File: .github/workflows/trivy-scan.yaml (hunk 1 above)
Comment (IsuruGunarathne) [reply .github/workflows/trivy-scan.yaml]: Updated in https://github.com/wso2/api-platform/pull/1246/commits/b011682c718440ebdaee0e476744579365a3cdb8

**Review by renuka-fernando — APPROVED** [review]
